File: Bukkit/src/main/java/me/mrkirby153/kcutils/scoreboard/items/ElementHeadedText.java
package me.mrkirby153.kcutils.scoreboard.items;

import me.mrkirby153.kcutils.scoreboard.ScoreboardElement;

public class ElementHeadedText implements ScoreboardElement {

    private ElementText header;
    private ElementText text;

    public ElementHeadedText(String header, String text) {
        this.header = new ElementText(header);
        this.text = new ElementText("   " + text);
    }

    @Override
    public String[] getLines() {
        String[] data = new String[2];
        data[0] = header.getLines()[0];
        data[1] = text.getLines()[0];
        return data;
    }
}

File: Bukkit/src/main/kotlin/me/mrkirby153/kcutils/scoreboard/items/ElementHeadedText.kt
package me.mrkirby153.kcutils.scoreboard.items

import me.mrkirby153.kcutils.scoreboard.ScoreboardElement

/**
 * A scoreboard element representing two lines: a header and the content
 */
class ElementHeadedText(header: String, text: String) : ScoreboardElement {

    /**
     * The header
     */
    private val header: ElementText = ElementText(header)

    /**
     * The text
     */
    private val text: ElementText = ElementText("   " + text)

    override val lines: Array<String>
        get() {
            return arrayOf(header.lines.first(), text.lines.first())
        }
}
